src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -295,8 +293,10 @@ private static Credentials newUsernamePasswordCredentials(
                 usernameData == null || usernameData.length()== 0 ||
                         passwordData == null || passwordData.length() == 0) {
             logger.log(Level.WARNING, "Invalid secret data, secretName: " +
-                secretName + " usernameData: " + usernameData + " passwordData: " +
-                passwordData);
+                secretName + " usernameData is null: " + (usernameData == null)
+                + " usernameData is empty: " + (usernameData.length() == 0) + 
+                " passwordData is null: " + (passwordData == null) + 
+                " passwordData is empty: " + (passwordData.length() == 0));
```

Comment:
won't this throw an NPE if passwordData==null?

Reply:
no the passwordData == null check just prior prevents that

once any of those conditions are true it goes in the if block and skips any additional conditionals

Reply:
this isn't an if block, it's a print statement.